libqtile/hook.py

```diff
@@ -914,7 +887,16 @@ def hooked_function():
           However, the same socket will need to be passed wherever you run ``qtile cmd-obj`` or ``qtile shell``.
 
         """
-        return self._user
+        if hook_name in dir(self):
+            raise ValueError(f"Do not use name of existing hook for custom hooks: {hook_name}.")
+
+        def _wrapper(func):
+            name = f"user_{hook_name}"
```

Comment:
Just nitpicking: maybe we could store these hooks somewhere else (.e.g `self.user_hooks`) to avoid that little trick?

Reply:
Not sure about that. We just have the one entry point for firing events (`hook.fire`) so you could still have a name clash even if the hooks were stored in different sets. Also, `hook.subscriptions` is a single dict with the hook name as the key. Without the prefix you could have clashes.

Reply:
Won't there be a single `hook.fire` call in the codebase for user hooks, the one in `Core.fire_user_hook`? Unless there are other potential use cases that might make it confusing, we could always add a `hook.fire_user` or something to select that other set?

np either way though :)

Reply:
My expectation with the user hooks is that they're fired by external apps and so need to be fired via the command interface.